### PR TITLE
Fix whitespace detector to handle deleted files.

### PR DESCRIPTION
**Pull request by gspencergoog**
The trailing whitespace detector wasn't handling file deletes very well (at all, really).

This filters the set of files grepped to only include files that exist.

Also, clarified the failure message to make it more obvious what the failure is when the grep finds results.

### CODE REVIEW

File: dev/bots/test.dart [hunk 1]
```diff
@@ -103,7 +103,6 @@ Future<Null> _checkForTrailingSpaces() async {
     final String commitRange = Platform.environment.containsKey('TEST_COMMIT_RANGE')
         ? Platform.environment['TEST_COMMIT_RANGE']
         : 'master..HEAD';
-    print('Checking for trailing whitespace in source files.');
     final List<String> fileTypes = <String>[
       '*.dart', '*.cxx', '*.cpp', '*.cc', '*.c', '*.C', '*.h', '*.java', '*.mm', '*.m',
```
Comment (Hixie) [inline dev/bots/test.dart]: also yaml?

**Comment by Hixie** [comment]
LGTM

File: dev/bots/test.dart [hunk 2]
```diff
@@ -115,8 +114,12 @@ Future<Null> _checkForTrailingSpaces() async {
       print('No Results for whitespace check.');
       return;
     }
+    // Only include files that actually exist, so that we don't try and grep for
+    // nonexistent files (can occur when files are deleted or moved).
     final List<String> changedFiles = changedFilesResult.stdout.trim().split('\n')
-        .where((String item) => item.trim().isNotEmpty).toList();
+        .where((String filename) {
+          return filename.trim().isNotEmpty && new File(filename).existsSync();
```
[the remark this answers is not in the document]
Comment (gspencergoog) [reply dev/bots/test.dart]: I suppose we don't have to.  I was just being paranoid that the git output might include indentation. Really, I don't have to do the "isNotEmpty" check either, now that I'm checking for existence.

File: dev/bots/test.dart (hunk 1 above)
Comment (gspencergoog) [reply dev/bots/test.dart]: Done.

**Review by liyuqian — APPROVED** [review]

**Pull request merged**